File: java/path_sum_112/Solution.java
package path_sum_112;

import tree.binary_tree.TreeNode;

public class Solution {
  public boolean hasPathSum(TreeNode root, int targetSum) {
    if (root == null) {
      return false;
    }

    if (root.left == null && root.right == null) {
      return targetSum == root.val;
    }

    return hasPathSum(root.left, targetSum - root.val)
        || hasPathSum(root.right, targetSum - root.val);
  }
}

File: c/path_sum_112/solution.c
#include <stdlib.h>

#include "solution.h"

bool hasPathSum(struct TreeNode* root, int targetSum) {
    if (root == NULL) {
        return false;
    }

    if (root->left == NULL && root->right == NULL) {
        return root->val == targetSum;
    }

    return hasPathSum(root->left, targetSum - root->val) 
    || hasPathSum(root->right, targetSum - root->val);
}